tests/unittest.py

```diff
@@ -155,20 +155,78 @@ def run_to_completion(self):
             self.clock.advance(amount)
 
     def assertTipsEqual(self, manager1, manager2):
-        s1 = set(manager1.tx_storage.get_all_tips())
-        s2 = set(manager2.tx_storage.get_all_tips())
+        assert (
+            hasattr(self, '_enable_sync_v1') and
+            hasattr(self, '_enable_sync_v2') and
+            (self._enable_sync_v1 or self._enable_sync_v2)
+        ), (
+            'Please set both `_enable_sync_v1` and `_enable_sync_v2` on the class. '
+            'Also they can\'t both be False. '
+            'This is by design so it\'s we don\'t forget to test for multiple sync versions.'
+        )
+        if self._enable_sync_v2:
+            self.assertTipsEqualSyncV2(manager1, manager2)
+        else:
+            self.assertTipsEqualSyncV1(manager1, manager2)
+
+    def assertTipsEqualSyncV1(self, manager1, manager2):
+        # tx tips
+        self.assertEqual(manager1.tx_storage._mempool_tips_index, manager2.tx_storage._mempool_tips_index)
+
+        # best block
+        s1 = set(manager1.tx_storage.get_best_block_tips())
+        s2 = set(manager2.tx_storage.get_best_block_tips())
         self.assertEqual(s1, s2)
 
-        s1 = set(manager1.tx_storage.get_tx_tips())
-        s2 = set(manager2.tx_storage.get_tx_tips())
+    def assertTipsEqualSyncV2(self, manager1, manager2):
+        # tx tips
+        tips1 = set(manager1.tx_storage._mempool_tips_index)
+        tips2 = set(manager2.tx_storage._mempool_tips_index)
+        self.log.debug('tx tips1', len=len(tips1), list=shorten_hash(tips1))
+        self.log.debug('tx tips2', len=len(tips2), list=shorten_hash(tips2))
+        # self.assertEqual(tips1, tips2)
+        # XXX: this is still not correct, tips may diverge since voided transactions aren't propagated
+        # XXX: temporarily disabled because tips may not match since sync-v2-mempool isn't merged yet
+        if len(tips1) < len(tips2):
```

Comment:
Maybe we should assert that all non-voided transactions are equal. What do you think?